source/MaterialXFormat/Util.cpp

```diff
@@ -234,6 +234,97 @@ void flattenFilenames(DocumentPtr doc, const FileSearchPath& searchPath, StringR
     }
 }
 
+FileSearchPath getResolvedDefinitionPath(const FileSearchPath& userDefinitionPath, bool includeSubFolders)
+{
+    FileSearchPath resolvedDefinitionPath;
+    const FileSearchPath coreDefinitionPath = getCoreDefinitionPath();
+    for (FileSearchPath::ConstIterator path=coreDefinitionPath.begin(); path != coreDefinitionPath.end(); ++path)
+    {
+        resolvedDefinitionPath.append(*path);
+    }
+    const FileSearchPath environmentPath = getEnvironmentPath();
+    for (FileSearchPath::ConstIterator path=environmentPath.begin(); path != environmentPath.end(); ++path)
+    {
+        resolvedDefinitionPath.append(*path);
+    }
+    for (FileSearchPath::ConstIterator path=userDefinitionPath.begin(); path != userDefinitionPath.end(); ++path)
+    {
+        resolvedDefinitionPath.append(*path);
+    }
+    const FileSearchPath assetDefinitionPath = getAssetDefinitionPath();
+    for (FileSearchPath::ConstIterator path=assetDefinitionPath.begin(); path != assetDefinitionPath.end(); ++path)
+    {
+        resolvedDefinitionPath.append(*path);
+    }
+    if (includeSubFolders)
+    {
+        std::set<std::string> childDefinitionPathsSet;
+        FileSearchPath childDefinitionPaths;
+        for (FileSearchPath::ConstIterator path = resolvedDefinitionPath.begin(); path != resolvedDefinitionPath.end(); ++path)
+        {
+            FilePath filePath = *path;
+            FilePathVec subdirs = filePath.getSubDirectories();
```

Comment:
Check if getSubDirectories() does not include the filePath, then please add this in first.

Reply:
I checked and it does include it.